var/spack/repos/builtin/packages/mkl/package.py

```diff
@@ -24,13 +24,38 @@ class Mkl(IntelInstaller):
     version('11.3.3.210', 'f72546df27f5ebb0941b5d21fd804e34',
             url="file://%s/l_mkl_11.3.3.210.tgz" % os.getcwd())
 
+    variant('shared', default=True, description='Builds shared library')
+    variant('ilp64', default=False, description='64 bit integers')
+    variant('openmp', default=False, description='OpenMP multithreading layer')
+
     # virtual dependency
     provides('blas')
     provides('lapack')
     # TODO: MKL also provides implementation of Scalapack.
 
-    def install(self, spec, prefix):
+    @property
+    def blas_libs(self):
+        shared = True if '+shared' in self.spec else False
+        suffix = dso_suffix if '+shared' in self.spec else 'a'
+        mkl_integer = ['libmkl_intel_ilp64'] if '+ilp64' in self.spec else ['libmkl_intel_lp64']  # NOQA: ignore=E501
+        mkl_threading = ['libmkl_intel_thread'] if '+openmp' in self.spec else ['libmkl_sequential']  # NOQA: ignore=E501
```

Comment:
minor issue, could you write here
```
if '~openmp~ in self.spec:
    mkl_threading = ['libmkl_sequential']
else:
   if '%intel' in self.spec:
       mkl_threading = ['libmkl_intel_thread' 'libiomp5']
  else:
      mkl_threading = ['libmkl_gnu_thread']
```
i.e. when threading is on, there are two libs for `intel` compiler and 1 for `gnu`.